Framework/PDFExporter/PDFExporter/Categories/UIKit/UITableView+PDFExporterDrawing.m

```diff
@@ -0,0 +1,125 @@
+    
+    [self drawBackgroundWithPath:path];
+    [self drawContentWithPath:path];
+    [self drawSubviewsWithPath:path withinPageRect:rect];
+    if (self.layer.borderWidth != 0) {
+        [self drawBorderWithPath:path];
+    }
+}
+
+- (void)drawSubviewsWithPath:(UIBezierPath *)path withinPageRect:(CGRect)rect {
+    CGPoint offset = CGPointZero;
+    for (CGFloat yCoordinate = CGRectGetMinY(rect);
+         yCoordinate < CGRectGetMaxY(rect);
+         yCoordinate += CGRectGetHeight(self.bounds)) {
+        offset.y = yCoordinate;
+        self.contentOffset = offset;
+        CGRect frame = self.frame;
+        [self layoutIfNeeded];
+        self.frame = frame;
+        [self layoutHeadersAndFooters];
+        [super drawSubviewsWithPath:path withinPageRect:rect];
+    }
+}
+
+- (BOOL)canDrawSubview:(UIView *)subview intersection:(CGRect)intersection {
+    if ([subview isKindOfClass:[UIScrollView class]]) {
+        return YES;
+    } else {
+        return [super canDrawSubview:subview intersection:intersection];
+    }
+}
+
+- (CGRect)subviewRect:(UIView *)subview drawingPageRect:(CGRect)rect {
+    if ([subview isKindOfClass:[UIScrollView class]]) {
+        return subview.drawingFrame; // hook for cells content view
+    } else {
+        return [super subviewRect:subview drawingPageRect:rect];
+    }
+}
+
+#pragma mark - Private
+
+- (void)layoutHeadersAndFooters {
+    for (NSUInteger section = 0; section < self.numberOfSections; ++section) {
+        UIView *headerView = [self headerViewForSection:section];
+        if (headerView) {
+            headerView.frame = [self rectForHeaderInSection:section];
+        }
+        UIView *footerView = [self footerViewForSection:section];
+        if (footerView) {
+            footerView.frame = [self rectForFooterInSection:section];
+        }
+    }
+}
+
+#pragma mark - PDFExporterPageInformation
+
+- (BOOL)canLayoutSubview:(UIView *)subview intersection:(CGRect)intersection {
+    if ([subview isKindOfClass:[UIScrollView class]]) {
+        return NO;
+    } else {
+        return [super canLayoutSubview:subview intersection:intersection];
+    }
+}
+
+- (BOOL)shouldConsiderLayoutSubview:(UIView *)subview intersection:(CGRect)intersection {
+    if ([subview isKindOfClass:[UIScrollView class]]) {
+        return YES;
+    } else {
+        return [super canLayoutSubview:subview intersection:intersection];
+    }
+}
+
+- (CGPoint)renderingOffsetForPageRect:(CGRect)rect {
+    CGPoint renderingOffset = CGPointZero;
+    CGPoint contentOffset = CGPointZero;
+    for (CGFloat yCoordinate = CGRectGetMinY(rect);
+         yCoordinate < CGRectGetMaxY(rect);
+         yCoordinate += CGRectGetHeight(self.bounds)) {
+        
+        // prepare view for rect section
+        contentOffset.y = yCoordinate;
+        self.contentOffset = contentOffset;
+        CGRect frame = self.frame;
+        [self layoutIfNeeded];
+        self.frame = frame;
+        [self layoutHeadersAndFooters];
```

Comment:
this chunk of code is duplicated and should be pulled out

Reply:
We have removed the duplicated code.